storage/gc_queue_test.go

```diff
@@ -314,6 +314,138 @@ func TestGCQueueProcess(t *testing.T) {
 	}
 }
 
+func TestGCQueueTransactionTable(t *testing.T) {
+	defer leaktest.AfterTest(t)
+
+	const now time.Duration = 3 * 24 * time.Hour
+	const dAbandon = -2 * DefaultHeartbeatInterval
+	type spec struct {
+		status      roachpb.TransactionStatus
+		ts          time.Duration
+		heartbeatTS time.Duration
+		intentSpans []roachpb.Span
+		seqGC       bool // expect sequence cache entries removed?
+	}
+	// Describes the state of the Txn table before the test.
+	before := map[string]spec{
```

Comment:
I would make all of these maps have keys of type `roachpb.Key` instead of casting from `string`. 

The before/after construct is awkward, with `seqGC` defined here and other expectations defined by changes made below. I would just add more "expectation" fields to the spec struct. And the use of `status = -1` could be replaced with a boolean since we don't do anything with this field in the `after` case except see whether it is -1.